### PR TITLE
InputHttp and OutputHttp implemented

### PR DESCRIPTION
**Pull request by AlexanderSuprunenko**

### CODE REVIEW

File: include/inout.h [hunk 1]
```diff
@@ -71,55 +74,151 @@
...
+        //The resulting value will be passed to mongoose
+        //So it is preferable that worker_action makes Output so that headers is empty and
+        //extra_headers is set to required complete value. This is for performance purpose.
+        //You can use combine_headers() to do this, like following
+        //  output.extra_headers = output.combine_headers();
+        //  output.headers.clear();
+        std::vector<std::pair<std::string, std::string>> headers;
+        std::string extra_headers;
+    private:
+        bool m_isHttp = false;
```
Comment (bitkis) [inline include/inout.h]: Bad name? How it's supposed to be used?

File: src/inout.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    { \
+        SET_STR_FLD(body); \
+        SET_STR_FLD(method); \
+        SET_STR_FLD(uri); \
+        SET_STR_FLD(proto); \
+        resp_code = hm.resp_code; \
+        SET_STR_FLD(resp_status_msg); \
+        SET_STR_FLD(query_string); \
+    }
+InOutHttpBase& InOutHttpBase::operator = (const http_message& hm)
```
Comment (bitkis) [inline src/inout.cpp]: Generally speaking, I'm not sure I like the idea of assigning objects of different and unrelated types to each other

File: src/inout.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#include "inout.h"
+#include "mongoose.h"
+
+namespace graft
+{
+std::unordered_map<std::string, std::string> OutHttp::uri_substitutions;
+
+//following #define depends on hm such as const http_message&
+#define SET_STR_FLD(fld) \
```
Comment (bitkis) [inline src/inout.cpp]: I see no stringnification or other tricks here... so is there a good reason to keep it a macros vs. making it a inline function?

File: src/inout.cpp (hunk 2 above)
Comment (AlexanderSuprunenko) [reply src/inout.cpp]: If we have constructor InOutHttpBase(const http_message& hm), it is natural to have assignment operator from the same type. Is not?

File: src/inout.cpp (hunk 3 above)
Comment (AlexanderSuprunenko) [reply src/inout.cpp]: There were tricks in my previous attempts, when it held mg_str like structures and set the same fields from objects of different types. But now it is not required. Have fixed.

File: src/inout.cpp (hunk 2 above)
Comment (AlexanderSuprunenko) [reply src/inout.cpp]: Good example, there are both 
basic_string& operator=( const CharT* s );
basic_string( const CharT* s, const Allocator& alloc = Allocator() );
What don't I understand? 

Maybe you want to say that they are not relates as string and char*. But
    using comp = std::complex<int>;
    comp a(1,2);
    //but
    comp c(1);
    c = 2;

Comment (bitkis) [reply src/inout.cpp]: OK, convinced :)

**Pull request merged**